test/e2e/app_management_test.go

```diff
@@ -429,14 +433,34 @@ func TestSyncResourceByLabel(t *testing.T) {
 		Sync().
 		Then().
 		And(func(app *Application) {
-			res, _ := fixture.RunCli("app", "sync", app.Name, "--label", fmt.Sprintf("app.kubernetes.io/instance=%s", app.Name))
-			assert.Contains(t, res, "guestbook-ui  Synced  Healthy")
-
-			res, _ = fixture.RunCli("app", "sync", app.Name, "--label", "this-label=does-not-exist")
+			_, _ = fixture.RunCli("app", "sync", app.Name, "--label", fmt.Sprintf("app.kubernetes.io/instance=%s", app.Name))
+		}).
+		Expect(SyncStatusIs(SyncStatusCodeSynced)).
+		And(func(app *Application) {
+			res, _ := fixture.RunCli("app", "sync", app.Name, "--label", "this-label=does-not-exist")
 			assert.Contains(t, res, "level=fatal")
 		})
 }
 
+func TestSyncAsync(t *testing.T) {
+	Given(t).
+		Path(guestbookPath).
+		When().
+		Create().
+		Sync().
+		Then().
+		And(func(app *Application) {
+
+			manipulateApplicationResources(t, app)
+
+			res, _ := fixture.RunCli("app", "sync", app.Name, "--async")
+			assert.Empty(t, res)
+
+			time.Sleep(2 * time.Second)
+		}).
+		Expect(SyncStatusIs(SyncStatusCodeSynced))
```

Comment:
You should probably have a call to `Expect(OperationPhaseIs(OperationSucceeded))` before this line.